src/profile/helpers/createActionAriaLabels.ts

```diff
@@ -0,0 +1,49 @@
+import { TFunction } from 'i18next';
+
+import { ActionAriaLabels } from '../components/editButtons/EditButtons';
+import { EditDataType } from './editData';
+
+export default function createActionAriaLabels(
+  dataType: EditDataType,
+  value: string,
+  t: TFunction
+): ActionAriaLabels {
+  if (dataType === 'phones' || dataType === 'emails') {
+    const labelForType =
+      dataType === 'phones'
+        ? t('profileInformation.phone')
+        : t('profileInformation.email');
+
+    const editTranslation =
+      dataType === 'phones'
+        ? t('profileInformation.ariaEditPhone')
+        : t('profileInformation.ariaEditEmail');
+
+    const primaryTranslation =
+      dataType === 'phones'
+        ? t('profileInformation.primaryPhoneAriaDescription')
+        : t('profileInformation.primaryEmailAriaDescription');
+
+    return {
+      edit: `${t(editTranslation)} ${value}`,
+      primary: `${t(primaryTranslation)}`,
+      setPrimary: `${t('profileForm.setPrimary')} ${labelForType} ${value}`,
+      remove: `${t('profileForm.remove')} ${labelForType} ${value}`,
+    };
+  }
+  if (dataType === 'addresses') {
+    return {
+      edit: `${t('profileInformation.ariaEditAddress')} ${value}`,
+      primary: `${t('profileInformation.primaryAddressAriaDescription')}`,
+      setPrimary: `${t('profileForm.setPrimary')} ${t(
+        'profileInformation.address'
+      )} ${value}`,
+      remove: `${t('profileForm.remove')} ${t(
+        'profileInformation.address'
+      )} ${value}`,
```

Comment:
Data should be inserted into the translations with place holders, not by concatenating strings in code. Nor should translated strings be concatenated together to form new user visible strings.